client/packages/system/src/Patient/PatientView/PatientView.tsx

```diff
@@ -161,10 +167,12 @@ export const PatientView: FC = () => {
     },
   ];
 
+  // Note: unmount modals when not used because they have some internal state that shouldn't be
+  // reused across calls.
   return (
     <React.Suspense fallback={<DetailViewSkeleton />}>
-      <ProgramDetailModal />
-      <EncounterDetailModal />
+      {current === PatientModal.Program ? <ProgramDetailModal /> : null}
```

Comment:
just a suggestion.. you could have each modal call `usePatientModalStore` and get `current` and return null, which would make this file prettier.

Or use a `isVisible` prop and pass in `current === PatientModal.Program` 🤷

Reply:
Thanks, it makes much more sense to have it there since the modal is where the problem occurs. [187e68d](https://github.com/openmsupply/open-msupply/pull/528/commits/187e68d6d0f53aa8a1f6430b208ba3c28a560a53)

Reply:
Just realized this would not unmount the component and I have the same problem :(